include/list.h

```diff
@@ -326,7 +326,7 @@ typedef struct xLIST
         }                                                                        \
                                                                                  \
         ( pxItemToRemove )->pxContainer = NULL;                                  \
-        ( pxList->uxNumberOfItems )--;                                           \
+        pxList->uxNumberOfItems -= 1U;                                           \
```

Comment:
pxList->uxNumberOfItems has the type _UBaseType_t_ . This operation leads to -Wconversion error , conversion from ‘unsigned int’ to ‘UBaseType_t’ . It can be changed to 
`pxList->uxNumberOfItems = ( UBaseType_t )( pxList->uxNumberOfItems - 1U );`